### DISCUSSION TITLE
Feedback: init

### DISCUSSION THREAD

**Discussion by ben-hamel** [mswjs/mswjs.io#340] [Q&A]
- [Open page](https://mswjs.io/docs/cli/init/)

## Feedback

I have msw.workerDirectory saved in my package json, I was looking at this issue - https://github.com/mswjs/msw/discussions/1015 and it sounds like it should auto-install `mockServiceWorker.js` when I run `npm install`. I can't get this to work, the only way to generate again is running `npx msw init ./public`.

As for the documentation, `whenever you install the msw package,` isn't super clear. Do you mean install completely again `npm install msw --save-dev ` or something else. Nothing seems to work other than `npx msw init ./public`. 

I would expect more of something like a code example or something to make it more clear. I'm trying to wrap my head around if I should store `mockServiceWorker.js` in git or not and it's a lot of work to get this figured out, would like a better explanation of how `msw.workerDirectory` works, what I need to do to trigger it and if I should commit `mockServiceWorker.js` or not. 

Thanks for your time,

- Ben

**Comment by kettanaito** [comment]
Hey, @ben-hamel. Thanks for raising this. 

> As for the documentation, whenever you install the msw package, isn't super clear.

MSW comes with a `postinstall` hook that will run an executable whenever you will install _MSW's dependencies_ (that's how the hook works, I believe). `npm install msw` will trigger that hook. `npm install` will also trigger that hook because all dependencies include MSW. 

The `msw.workerDirectory` option is meant to update the worker script's contents when you _install different versions of MSW_ (not necessarily the latest, maybe you wish to roll back). This implies that you run `npm install msw@x.y.z`. Is that what you are doing? 

> I'm trying to wrap my head around if I should store mockServiceWorker.js in git

I think you should. I lean towards the philosophy that you own that worker script since it lives in your `public` directory. We may also consider migrating away from the `postinstall` script since it's generally undesired for third-parties (see https://github.com/mswjs/msw/issues/1297).

**Comment by ben-hamel** [comment]
okay, Thanks for clarifying this! So I must be having an issue with MSW not running the post-install script. I'll test a bit more and log an issue if needed.

**Reply by kettanaito** [reply]
Note that it's not MSW that runs `postinstall` but your package manager (npm, yarn, pnpm). Different package managers have different behaviors. Perhaps that's the root cause. Let me know if you find out anything!
